tensorflow_federated/python/core/impl/executors/federating_executor.py

```diff
@@ -453,6 +451,221 @@ def validate_executor_placements(cls, executor_placements):
+    # to `federated_reduce()`. The more complete implementation would be able
+    # to take advantage of the parallelism afforded by `merge` to reduce the
+    # cost from liner (with respect to the number of clients) to sub-linear.
+
+    # TODO(b/134543154): Expand this implementation to take advantage of the
+    # parallelism afforded by `merge`.
+    fed_ex = self.federating_executor
+
+    val = arg.internal_representation[0]
+    zero = arg.internal_representation[1]
+    accumulate = arg.internal_representation[2]
+    pre_report = await fed_ex._compute_intrinsic_federated_reduce(
+        FederatingExecutorValue(
+            anonymous_tuple.AnonymousTuple([(None, val), (None, zero),
+                                            (None, accumulate)]),
+            computation_types.NamedTupleType(
+                (val_type, zero_type, accumulate_type))))
+
+    py_typecheck.check_type(pre_report.type_signature,
+                            computation_types.FederatedType)
+    type_utils.check_equivalent_types(pre_report.type_signature.member,
+                                      report_type.parameter)
+
+    report = arg.internal_representation[4]
+    return await fed_ex._compute_intrinsic_federated_apply(
+        FederatingExecutorValue(
+            anonymous_tuple.AnonymousTuple([
+                (None, report), (None, pre_report.internal_representation)
+            ]),
+            computation_types.NamedTupleType(
+                (report_type, pre_report.type_signature))))
+
+  async def federated_sum(self, arg):
+    py_typecheck.check_type(arg.type_signature, computation_types.FederatedType)
+    zero, plus = tuple(await asyncio.gather(*[
+        executor_utils.embed_tf_scalar_constant(
+            self.federating_executor,
+            arg.type_signature.member,
+            0),
+        executor_utils.embed_tf_binary_operator(
+            self.federating_executor,
+            arg.type_signature.member,
+            tf.add)]))
+    return await self.federating_executor._compute_intrinsic_federated_reduce(
+        FederatingExecutorValue(
+            anonymous_tuple.AnonymousTuple([
+                (None, arg.internal_representation),
+                (None, zero.internal_representation),
+                (None, plus.internal_representation)
+            ]),
+            computation_types.NamedTupleType(
+                (arg.type_signature, zero.type_signature, plus.type_signature)))
+    )
+
+  async def federated_mean(self, arg):
+    # TODO check this logic, just switched SERVER to AGGREGATOR
+    arg_sum = await self.federating_executor._compute_intrinsic_federated_sum(arg)
+    member_type = arg_sum.type_signature.member
+    count = float(len(arg.internal_representation))
+    if count < 1.0:
+      raise RuntimeError('Cannot compute a federated mean over an empty group.')
+    child = self._get_child_executors(placement_literals.AGGREGATOR, index=0)
```

Comment:
don't know if there's a good solution but it could lead to some confusion that in `validate_executor_placements` we're explicitly checking that only a single aggregator is passed in, yet here we ask for the first one due to the list wrapping in `FederatingExecutor.__init__`

Reply:
That's the original code's way of doing it, I just wrapped their logic in a method call.

Reply:
I get what you mean though, there's some semantic weirdness around plurality with these names